### ISSUE TITLE
Sheldon#list does not detect broken symlinks

### ISSUE THREAD

**Issue by dvjones89** [dvjones89/sheldon#13]
Currently, Sheldon checks for the presence of a symlink, but does not verify whether the symlink actually points to a real file on the system.
In other words, a dead symlink still appears green in Sheldon's coloured output.

Detecting these dead symlinks would make Sheldon#list more reliable, and would likely be helpful when implementing Sheldon#forget (see issue #4 )

**Comment by dvjones89** [comment]
Looks like this is the way to fix the issue:

```ruby
File.exists?(File.readlink("/Users/dave/.zshrc"))
=> true

File.exists?(File.readlink("/Users/dave/.test"))
=> false
```

**Comment by dvjones89** [comment]
Fixed in 1d965087a83ea406a706dbfce67c126c593d654b
